server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -93,13 +103,22 @@ public MailboxFactory(MailboxManager mailboxManager) {
         return name;
     }
 
-    @VisibleForTesting Optional<MailboxId> getParentIdFromMailboxPath(MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException {
+    @VisibleForTesting Optional<MailboxId> getParentIdFromMailboxPath(MailboxPath mailboxPath, Optional<List<MailboxMetaData>> userMailboxesMetadata,
```

Comment:
For me it seems we could have use `List<MailboxMetaData>` instead of `Optional<List<MailboxMetaData>>` the empty case would be a empty list. No ?